packages/react-dom/src/events/accumulateTwoPhaseListeners.js

```diff
@@ -31,20 +31,22 @@ export default function accumulateTwoPhaseListeners(
   const phasedRegistrationNames = event.dispatchConfig.phasedRegistrationNames;
   const dispatchListeners = [];
   const dispatchInstances = [];
+  const dispatchCurrentTargets = [];
 
   const {bubbled, captured} = phasedRegistrationNames;
   // If we are not handling EventTarget only phase, then we're doing the
   // usual two phase accumulation using the React fiber tree to pick up
   // all relevant useEvent and on* prop events.
-  let node = event._targetInst;
+  let instance = event._targetInst;
   let lastHostComponent = null;
 
   // Accumulate all instances and listeners via the target -> root path.
-  while (node !== null) {
-    const {stateNode: instance, tag} = node;
+  while (instance !== null) {
+    const {stateNode, tag} = instance;
     // Handle listeners that are on HostComponents (i.e. <div>)
-    if (instance !== null && tag === HostComponent) {
-      lastHostComponent = instance;
+    if (stateNode !== null && tag === HostComponent) {
```

Comment:
Let's make tag checks first here and elsewhere. We don't care about other state nodes anyway.